src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/task.py

```diff
@@ -349,12 +355,80 @@ def acr_task_update(cmd,  # pylint: disable=too-many-locals
         trigger=TriggerUpdateParameters(
             source_triggers=source_trigger_update_params,
             base_image_trigger=base_image_trigger_update_params
+        ),
+        credentials=get_custom_registry_credentials(
+            cmd=cmd,
+            auth_mode=auth_mode
         )
     )
 
     return client.update(resource_group_name, registry_name, task_name, taskUpdateParameters)
 
 
+def acr_task_credential_add(cmd,
+                            client,
+                            task_name,
+                            registry_name,
+                            login_server,
+                            username,
+                            password,
+                            resource_group_name=None):
+    _, resource_group_name = validate_managed_registry(
+        cmd, registry_name, resource_group_name, TASK_NOT_SUPPORTED)
+
+    TaskUpdateParameters = cmd.get_models('TaskUpdateParameters')
+    taskUpdateParameters = TaskUpdateParameters(
+        credentials=get_custom_registry_credentials(
+            cmd=cmd,
+            login_server=login_server,
+            username=username,
+            password=password
+        )
+    )
+
+    resp = LongRunningOperation(cmd.cli_ctx)(
+        client.update(resource_group_name, registry_name, task_name, taskUpdateParameters)
+    )
+    resp = resp.credentials
+    return {} if not resp else resp.custom_registries
+
+
+def acr_task_credential_remove(cmd,
+                               client,
+                               task_name,
+                               registry_name,
+                               login_server,
+                               resource_group_name=None):
+    _, resource_group_name = validate_managed_registry(
+        cmd, registry_name, resource_group_name, TASK_NOT_SUPPORTED)
+
+    TaskUpdateParameters = cmd.get_models('TaskUpdateParameters')
+    taskUpdateParameters = TaskUpdateParameters(
+        credentials=get_custom_registry_credentials(
+            cmd=cmd,
+            login_server=login_server
+        )
+    )
+
+    resp = LongRunningOperation(cmd.cli_ctx)(
+        client.update(resource_group_name, registry_name, task_name, taskUpdateParameters)
+    )
+    resp = resp.credentials
+    return {} if not resp else resp.custom_registries
+
+
+def acr_task_credential_list(cmd,
+                             client,
+                             task_name,
+                             registry_name,
+                             resource_group_name=None):
+    _, resource_group_name = validate_managed_registry(
+        cmd, registry_name, resource_group_name, TASK_NOT_SUPPORTED)
+
+    resp = client.get_details(resource_group_name, registry_name, task_name).credentials
+    return {} if not resp else resp.custom_registries
```

Comment:
@mnltejaswini Why are we returning null for empty dict?

Reply:
@djyou I created a workitem for this and we will return empty dict. This code handles both null and {}